<?xml version="1.0" encoding="utf-8"?>

<launch>
	<node ns="camera1" pkg="image_proc" type="image_proc" name="image_procg" />
	<node ns="camera1" pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera" output="screen">
		<param name="width" type="int" value="1280" />
		<param name="height" type="int" value="1024" />
		<param name="fps" type="int" value="30" />
		<param name="frame_id" type="string" value="/camera1" />
		<param name="device" type="string" value="/dev/video0" />
		<param name="camera_info_url" type="string" value="file://$(find ar_sys)/camera_info/camera_calibration.yaml" />
	</node>

	<node pkg="ar_sys" type="multi_boards" name="ar_multi_boards" output="screen">
		<remap from="/camera_info" to="/camera1/camera_info" />
		<remap from="/image" to="/camera1/image_rect_color" />

		<param name="image_is_rectified" type="bool" value="true"/>
		<param name="boards_config" type="string" value="$(find ar_sys)/data/multi/boards.yml"/>
		<param name="boards_directory" type="string" value="$(find ar_sys)/data/multi"/>
		<param name="draw_markers" type="bool" value="false" />
		<param name="draw_markers_cube" type="bool" value="true" />
		<param name="draw_markers_axis" type="bool" value="false" />
	</node>

	<node pkg="image_view" type="image_view" name="image_view" output="screen">
		<remap from="image" to="/ar_multi_boards/result" />

		<param name="autosize" type="bool" value="true" />
	</node>

</launch>

